app/uk/gov/hmrc/serviceconfigs/controller/DeploymentConfigController.scala

```diff
@@ -47,6 +49,26 @@ class DeploymentConfigController @Inject()(
         })
     }
 
+  def deploymentConfigGrouped(serviceName: Option[ServiceName], teamName: Option[TeamName], repoType: Option[String], sort: Option[String]): Action[AnyContent] =
+    Action.async {
+
+      val sn: Option[ServiceName] = serviceName.map(_.trimmed).filter(_.asString.nonEmpty)
```

Comment:
I think this is unnecessary, `?serviceName=` already has ambiguous semantics (can look like it's saying where serviceName is empty) and if a client provided `?serviceName=%20` - I think it should be respected - the client should not provide the filter when not required.